Sources/Sentry/SentryHttpTransport.m

```diff
@@ -293,7 +292,9 @@ - (SentryEnvelope *)addClientReportTo:(SentryEnvelope *)envelope
         [self.discardedEvents removeAllObjects];
     }
 
-    SentryClientReport *clientReport = [[SentryClientReport alloc] initWithDiscardedEvents:events];
+    SentryClientReport *clientReport = [[SentryClientReport alloc]
+        initWithDiscardedEvents:events
+                   dateProvider:SentryDependencyContainer.sharedInstance.dateProvider];
```

Comment:
`m`: We already have a property for the date provider. No need to use the `SentryDependencyContainer` here.
```suggestion
                   dateProvider:self.dateProvider];
```